src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/tests/stores/memoryFormStoreFactory.test.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import MemoryFormStore from '../../stores/MemoryFormStore';
+import memoryFormStoreFactory from '../../stores/memoryFormStoreFactory';
+import metadataStore from '../../stores/metadataStore';
+import SchemaFormStore from '../../stores/SchemaFormStore';
+
+jest.mock('../../stores/metadataStore', () => ({
+    getJsonSchema: jest.fn(),
+    getSchema: jest.fn(),
+}));
+
+test('Create a MemoryFormStore with schema', (done) => {
+    const schema = {
+        title: {},
+    };
+    const jsonSchema = {
+        schema: {},
+    };
+
+    const schemaPromise = Promise.resolve(schema);
+    const jsonSchemaPromise = Promise.resolve(jsonSchema);
+
+    metadataStore.getSchema.mockReturnValue(schemaPromise);
+    metadataStore.getJsonSchema.mockReturnValue(jsonSchemaPromise);
+
+    const memoryFormStore = memoryFormStoreFactory.createFromFormKey('test');
+
+    expect(memoryFormStore).toBeInstanceOf(SchemaFormStore);
+    setTimeout(() => {
```

Comment:
did you use `setTimeout` instead of `return Promise.all([schemaPromise, jsonSchemaPromise]).then(() => {` on purpose here?

Reply:
Yeah, `Promise.all` was not working, but I was playing around quite a bit, maybe it does work now without it... Will double-check it.